WCF/Shared/Implementation/WcfOperationContext.cs

```diff
@@ -115,7 +129,7 @@ private static IOperationContext GetContext()
             {
                 if ( owner != null )
                 {
-                    context = new WcfOperationContext(owner);
+                    context = new WcfOperationContext(owner, HttpContext.Current);
```

Comment:
@tomasr, the key here is to make it only once, not on every request. This may help with performance.

One way to do it will be to check [HttpRuntime.IISVersion](https://msdn.microsoft.com/en-us/library/system.web.httpruntime.iisversion(v=vs.110).aspx) on start of SDK and if it returns nothing – do not use `HttpContext.Current` to initialize the operation.